File: Assets/Custom Assets/Scripts/EnemyHpBar.js
#pragma strict

var target : Transform; 
var cam : Camera;
var Jugador: GameObject;

function Start () {
	Debug.Log("Se crea Barra " + GetInstanceID());
}

function Update () 
{ 
	if(!Jugador){
		Jugador = GameObject.FindGameObjectWithTag("Player");
		cam = Jugador.transform.FindChild("GameCamera").camera;
	}

	if(target != null)
	{
	var wantedPos = cam.WorldToViewportPoint (target.position); 
  	transform.position = wantedPos; 	
	}
  
}

function DestroyBar()
{
	Debug.Log("Destruyendo Barra " + GetInstanceID());
	Destroy(gameObject);
}

File: Assets/Custom Assets/Scripts/EnemyHpBar.cs
using UnityEngine;

public class EnemyHpBar : MonoBehaviour
{
    public Transform target;
    public Camera cam;
    public GameObject Jugador;

    private void Start()
    {
        Debug.Log("Se crea Barra " + GetInstanceID());
    }

    private void Update()
    {
        if (!Jugador)
        {
            Jugador = GameObject.FindGameObjectWithTag("Player");
            cam = Jugador.transform.Find("GameCamera").GetComponent<Camera>();
        }

        if (target != null)
        {
            Vector3 wantedPos = cam.WorldToViewportPoint(target.position);
            transform.position = wantedPos;
        }
    }

    public void DestroyBar()
    {
        Debug.Log("Destruyendo Barra " + GetInstanceID());
        Destroy(gameObject);
    }
}
